src/main/java/jenkins/plugins/git/AbstractGitSCMSource.java

```diff
@@ -384,11 +384,12 @@ public SCMRevision run(GitClient client, String remoteName) throws IOException,
                                           // tag does not exist
                                           return null;
                                       }
-                                  } else if (head instanceof GitBranchSCMHead) {
+                                  } else if (head instanceof GitBranchSCMHead || !(head instanceof GitSCMHeadMixin)) {
+                                      GitBranchSCMHead branchHead = (head instanceof GitBranchSCMHead) ? (GitBranchSCMHead)head : new GitBranchSCMHead(head.getName());
```

Comment:
Hmmm I think we need to return an old-style `AbstractGitSCM.RevisionImpl` if it is `!(head instanceof GitSCMHeadMixin)` rather than swap out the head for another head because then the implicit contract that `assert revision == null || head.equals(revision.getHead())` on the return value would be broken

Reply:
Then I could revert and just let the `else` block below catch it?

Reply:
perhaps... I can't see the diff easily in this browser